android-json-form-wizard/src/main/java/com/vijay/jsonwizard/activities/JsonFormActivity.java

```diff
@@ -372,39 +365,57 @@ public View getFormDataView(String address) {
     @Override
     public JSONObject getObjectUsingAddress(String[] address, boolean popup) throws JSONException {
         if (address != null && address.length > 1) {
-            if (RuleConstant.RULES_ENGINE.equals(address[0])) {
+            if (RuleConstant.RULES_ENGINE.equals(address[0]) && !address[1].endsWith(".yml")) {
+                JSONArray jsonArray = new JSONArray(address[1]);
+                List<String> keysList = new ArrayList<>();
+                //index 0 has the key
+                for (int i = 1; i < jsonArray.length(); i++) {
+                    JSONObject jsonObject = jsonArray.optJSONObject(i);
+                    keysList.addAll(Utils.getConditionKeys(jsonObject.optString(RuleConstant.CONDITION)));
```

Comment:
Shouldn't this only happen if  `jsonObject.optString(RuleConstant.CONDITION)` returns a non-empty string?

Reply:
if the condition is empty then the rules where defined incorrectly.....its more of user input issue